.github/workflows/docker-build-test.yaml

```diff
@@ -61,6 +61,10 @@ env:
   # We use `pr-<pr_number>` as cache-id for PRs and simply <branch_name> otherwise.
   TARGET_CACHE_ID: ${{ github.event.number && format('pr-{0}', github.event.number) || github.ref_name }}
 
+  # On PRs, only build and push to GCP
+  # On push, build and push to all remote registries
+  TARGET_REGISTRY: ${{ github.event_name == 'pull_request_target' && 'remote' || 'remote-all' }}
```

Comment:
NOTE: to make this change backwards compatible, and not require all pending PRs to rebase, I've made the default `TARGET_REGISTRY` on PRs the same as before -- `remote`. `remote` will exist as deprecated, but it will essentially have the same function as setting the value as `gcp`.